control-plane/pkg/security/secrets_provider_net_spec.go

```diff
@@ -53,21 +56,27 @@ func ResolveAuthContextFromNetSpec(lister corelisters.SecretLister, namespace st
 			return nil, err
 		}
 	}
-	references, virtualSecretData := toContract(securityFields)
+	references, virtualSecretData, virtualSecretName, virtualSecretNamespace := toContract(securityFields)
 	multiSecretReference := &contract.MultiSecretReference{
 		Protocol:   getProtocolContractFromNetSpec(netSpec),
 		References: references,
 	}
 	virtualSecretData[ProtocolKey] = []byte(getProtocolFromNetSpec(netSpec))
 
 	authContext := &NetSpecAuthContext{
-		VirtualSecret:        &corev1.Secret{Data: virtualSecretData},
+		VirtualSecret: &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      virtualSecretName,
+				Namespace: virtualSecretNamespace,
+			},
+			Data: virtualSecretData,
+		},
 		MultiSecretReference: multiSecretReference,
 	}
 	return authContext, nil
 }
 
-func toContract(securityFields []*securityField) ([]*contract.SecretReference, map[string][]byte) {
+func toContract(securityFields []*securityField) ([]*contract.SecretReference, map[string][]byte, string, string) {
```

Comment:
can't we just secret as return value (data+name+namespace) ? Too many return values is hard to read and easy to swap wrong values (like name in place of namespace)